sjb/hack/determine_install_upgrade_version.py

```diff
@@ -96,7 +102,10 @@ def sort_pkgs(available_pkgs):
 	#          ORIGIN_UPGRADE_RELEASE_MINOR_VERSION=5
 	#
 	if args.dependency_branch.startswith("release") or args.dependency_branch.startswith("enterprise"):
-		pkg_version = args.dependency_branch.split("-")[1]
+		input_pkg_version_parsed = pkg_version.split(".")
+		dependency_branch_version_parsed = args.dependency_branch.split("-")[1].split(".")
+		dependency_branch_version_parsed[0] = input_pkg_version_parsed[0]
```

Comment:
Add this to the comment above the `if` and use explanatory variables to make it clearer:
```py
if args.dependency_branch.startswith("release") or args.dependency_branch.startswith("enterprise"):
    pkg_major_version = pkg_version.split(".")[0]
    dependency_minor_version = args.dependency_branch.split("-")[1].split(".")[1]
    pkg_version = ".".join(pkg_major_version, dependency_minor_version)
```